File: C++ Programming/Competitive Programming/Coding Challenges 1/program4.cpp
/*
4. Accept number of rows and number of columns from user and display below pattern.
Input : iRow = 6 iCol = 5
Output :* * * * *
        * @ @ @ *
        * @ @ @ *
        * @ @ @ *
        * @ @ @ *
        * * * * *
*/

#include<iostream>
using namespace std;

void Pattern(int iRow,int iCol)
{
    int i=1;
    int j=0;

    while (i <= iRow)
    {
        j = 1;
        while ( j <= iCol)
        {
            if (i > 1 && i < iRow && j > 1 && j < iCol  )
            {
                  cout<<"@ ";
            }
            else
            {
                 cout<<"* ";
            }
            j++;
        }
         cout<<"\n";
        i++;
    }
    
}

int main()
{
    int iValue1=0,iValue2=0;

    cout<<"Enter The Number of Row: ";
    cin>>iValue1;

    cout<<"Enter The Number of Column: ";
    cin>>iValue2;

    Pattern(iValue1,iValue2);

    return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 7/program4.java
/*

*/

import java.util.*;

class program4
{
    public static void Pattern(int iRow, int iCol)
    {
        int i = 1;
        int j = 1;

        do
        {
            j = 1;

            do
            {
                if (i == 1 || i == iRow || j == 1 || j == i || j == iCol)
                {
                    System.out.print("*\t");
                }
                else
                {
                    System.out.print(" \t");
                }

                j++;
            }
            while (j <= iCol);

            System.out.println();

            i++;
        }
        while (i <= iRow);
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0;
        int iValue2 = 0;

        System.out.println("Enter number of rows :");
        iValue1 = sobj.nextInt();

        System.out.println("Enter number of columns :");
        iValue2 = sobj.nextInt();

        Pattern(iValue1, iValue2);

        sobj.close();
    }
}